wlr_output_commit sometimes start to fail a lot

Tested on [current master](cbd4a2807bcfa74cc87c36d7ec20cdf92a7c77fc) the rendering "freezes".

My setup consists of two outputs. One 1080p laptop display and an external 4k output connected via displayport.

I did the following:
1. Open river
2. Spawn a bunch of terminals (foot)
3. Change the layout a couple of times to cause calls to `root.startTransaction`
4. Now changing the layout did not result in a visual change
5. Change focus to my other output and spawn a terminal. I didn't see any visual change here either
6. Change VT back and fourth
7. Now the rendering works as expected

I can kind of reliably reproduce this. The steps above lead to this most of the time.

Here are the logs: https://paste.sr.ht/~maringuu/bb42efa93eeed27b3d8d53258c907ee9f2771f87

Here are some other logs where the rendering also froze: 
https://paste.sr.ht/~maringuu/862ccc42f315baeeadffcd20fe667ad604027689


Edit:
I figured out a simpler way to reproduce. It works very reliable for me.
1. Open river
2. Open at least two terminals (foot)
3. Do `riverctl swap next` a bunch of times until it freezes

Is there some news about this? 
I also have this issue frequently, it happen on river and dwl but not on sway
I only have one screen, a laptop 1920x1080, no other monitors 
```
$ lspci | rg Graph  
00:02.0 VGA compatible controller: Intel Corporation Skylake GT2 [HD Graphics 520] (rev 07)
```
Nothing in `journalctl -p 3..1 -xb`

River log:
```
00:00:37.309 [ERROR] [backend/drm/atomic.c:36] eDP-1: Atomic commit failed (pageflip): Device or resource busy
err: (render) output commit failed for eDP-1
```

--- a/river/render.zig
+++ b/river/render.zig
@@ -143,8 +143,12 @@ pub fn renderOutput(output: *Output) void {
     }
 
     // TODO: handle failure
-    output.wlr_output.commit() catch
+    if (output.wlr_output.commit()) |_| {
+        return;
+    } else |_| {
+        renderOutput(output);
         log.err("output commit failed for {}", .{output.wlr_output.name});
+    }
 }
 
 fn renderFilter(view: *View, filter_tags: u32) bool {

--- a/river/render.zig
+++ b/river/render.zig
@@ -143,7 +143,7 @@ pub fn renderOutput(output: *Output) void {
     }
 
     output.wlr_output.commit() catch {
-        log.err("output commit failed for {}, retrying", .{output.wlr_output.name});
+        log.err("output commit failed for {}, retrying", .{std.mem.spanZ(&output.wlr_output.name)});
         output.wlr_output.scheduleFrame();
     };
 }

--- a/river/render.zig
+++ b/river/render.zig
@@ -142,7 +142,6 @@ pub fn renderOutput(output: *Output) void {
         output.wlr_output.setDamage(&damage);
     }
 
-    // TODO: handle failure
     output.wlr_output.commit() catch {
         log.err("output commit failed for {}, retrying", .{output.wlr_output.name});
         output.wlr_output.scheduleFrame();

--- a/river/render.zig
+++ b/river/render.zig
@@ -143,12 +143,10 @@ pub fn renderOutput(output: *Output) void {
     }
 
     // TODO: handle failure
-    if (output.wlr_output.commit()) |_| {
-        return;
-    } else |_| {
-        renderOutput(output);
-        log.err("output commit failed for {}", .{output.wlr_output.name});
-    }
+    output.wlr_output.commit() catch {
+        log.err("output commit failed for {}, retrying", .{output.wlr_output.name});
+        output.wlr_output.scheduleFrame();
+    };
 }
 
 fn renderFilter(view: *View, filter_tags: u32) bool {

